src/expr/term_conversion_proof_generator.cpp

```diff
@@ -121,13 +154,21 @@ std::shared_ptr<ProofNode> TConvProofGenerator::getProofForRewriting(Node t)
       if (!rcur.isNull())
       {
         // d_proof should have a proof of cur = rcur. Hence there is nothing
-        // to do here, as pf will reference prg to get the proof from d_proof.
-        // It may be the case that rcur also rewrites, thus we cannot assign
-        // the final rewritten form for cur yet. Instead we revisit cur after
-        // finishing visiting rcur.
-        rewritten[cur] = rcur;
-        visit.push_back(cur);
-        visit.push_back(rcur);
+        // to do here, as pf will reference d_proof to get its proof.
+        if (d_policy == TConvPolicy::FIXPOINT)
+        {
+          // It may be the case that rcur also rewrites, thus we cannot assign
+          // the final rewritten form for cur yet. Instead we revisit cur after
+          // finishing visiting rcur.
+          rewritten[cur] = rcur;
+          visit.push_back(cur);
+          visit.push_back(rcur);
+        }
+        else
```

Comment:
Just for safety, check that policy is `ONCE`, in case we add more in the future and to guarantee this is not forgotten.